python/tach/interactive/modules.py

```diff
@@ -427,6 +470,16 @@ def _(event: KeyPressEvent):
                 # Root should not be explicitly selected
                 return
             self.selected_node.is_module = not self.selected_node.is_module
+            if self.selected_node.is_module and self.selected_node.is_source_root:
+                # If we are marking the currently selected source root as a module,
+                # we should reset the source root to the root path.
+                # The source root cannot be a module.
+                self.file_tree.set_source_root(Path("."))
```

Comment:
Makes sense, I wonder if setting source root as parent vs root path could make more sense, but would have to play with it a bit more. I guess we could kind of figure it out by looking at your imports but that's way more work; this is def a sane default.